### DISCUSSION TITLE
Play a random audio with bevy 0.11 audio system

### DISCUSSION THREAD

**Discussion by aligator** [bevyengine/bevy#10007] [Q&A] [answered]
Hi, 

I am just doing a bevy 0.10 tutorial and adapt it to 0.11 on the fly.  
But I stumbled across a problem:  
Here the sound is just loaded and played based on a random number.
https://www.youtube.com/watch?v=4TjEo-gDgAg&t=311s

How would I do something like this in an ECS idiomatic way with the new 0.11 Audio system?

**Comment by rparrett** [comment, marked as answer]
Playing an audio clip in Bevy 0.11 is as easy as spawning an [`AudioBundle`](https://docs.rs/bevy/latest/bevy/audio/type.AudioBundle.html).

Here's some code similar to the YouTube video:

```rust
fn play_sound(mut commands: Commands, asset_server: Res<AssetServer>) {
    let sound = if random::<f32>() > 0.5 {
        asset_server.load("one.ogg")
    } else {
        asset_server.load("two.ogg")
    };

    commands.spawn(AudioBundle {
        source: sound,
        // auto-despawn the entity when playback finishes
        settings: PlaybackSettings::DESPAWN,
    });
}
```

There is one slight problem with this code: The audio files may be unloaded and their handles cleaned up by the asset server. This won't prevent things from working, but your app might have to do extra work when playing sounds which could introduce latency.

You can store their handles in a `Resource` to prevent this from happening:

```rust
// Be sure to initialize this resource when building your app in `fn main`
// with `.init_resource::<SoundHandles>`.
#[derive(Resource)]
struct SoundHandles {
    one: Handle<AudioSource>,
    two: Handle<AudioSource>,
}

impl FromWorld for SoundHandles {
    fn from_world(world: &mut World) -> Self {
        let asset_server = world.resource::<AssetServer>();
        Self {
            one: asset_server.load("audio/one.ogg"),
            two: asset_server.load("audio/two.ogg"),
        }
    }
}

fn play_sound(mut commands: Commands, handles: Res<SoundHandles>) {
    // Also, `random` is generic and will produce a `bool` that is `true` 50% of the time if used in this context.
    let sound = if random() {
        handles.one.clone()
    } else {
        handles.two.clone()
    };

    commands.spawn(AudioBundle {
        source: sound,
        // auto-despawn the entity when playback finishes
        settings: PlaybackSettings::DESPAWN,
    });
}
```

**Reply by aligator** [reply]
Nice, now I understood it. Many thanks :+1:
